components/GameScreen.vue

```diff
@@ -106,6 +110,130 @@ watch(gameState, async (newState, prevState) => {
       autoDensity: true,
     });
 
+    let isZoomingOut = false;
+    const zoomDuration = 500; // Duration of the zoom-out effect in milliseconds
+    let startZoomTime: number | null = null;
+    let startMousePos = { x: 0, y: 0 };
+
+    // Functions to handle smooth zoom out
+    function smoothZoomOut(mousePos: { x: number; y: number }) {
+      if (!appRef.value) {
+        return;
+      }
+
+      isZoomingOut = true;
+      startZoomTime = performance.now();
+      startMousePos = mousePos;
+      requestAnimationFrame(animateZoomOut);
+    }
+
+    function animateZoomOut(currentTime: number) {
+      if (!appRef.value || !startZoomTime) {
+        return;
+      };
+
+      const elapsedTime = currentTime - startZoomTime;
+      const progress = Math.min(elapsedTime / zoomDuration, 1);
+      const newScale = minZoom + (appRef.value.stage.scale.x - minZoom) * (1 - progress);
+
+      const worldPos = {
+        x: (startMousePos.x - appRef.value.stage.position.x) / appRef.value.stage.scale.x,
+        y: (startMousePos.y - appRef.value.stage.position.y) / appRef.value.stage.scale.y,
+      };
+
+      appRef.value.stage.scale.set(newScale);
+
+      const newScreenPos = {
+        x: worldPos.x * newScale + appRef.value.stage.position.x,
+        y: worldPos.y * newScale + appRef.value.stage.position.y,
+      };
+
+      appRef.value.stage.position.set(
+        Math.min(0, Math.max(appRef.value.stage.position.x - (newScreenPos.x - startMousePos.x), appRef.value.screen.width - appRef.value.screen.width * newScale)),
+        Math.min(0, Math.max(appRef.value.stage.position.y - (newScreenPos.y - startMousePos.y), appRef.value.screen.height - appRef.value.screen.height * newScale)),
+      );
+      if (progress < 1) {
+        requestAnimationFrame(animateZoomOut);
+      } else {
+        isZoomingOut = false;
+      }
+    }
+
+    // Mouse wheel event listener
+    canvas.value?.addEventListener("wheel", (event) => {
```

Comment:
Although today, this is being called once per canvas, it's a bit concerning that we don't have a logic to `.removeEventListener` because this initialization logic can potentially be called multiple times for the same canvas.

Maybe we can add a comment calling this out to make future refactors easier.

Reply:
Thank you!